bindings/c/src/lib.rs

```diff
@@ -301,6 +302,55 @@ pub unsafe extern "C" fn libsql_connect(
     0
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn libsql_load_extension(
+    conn: libsql_connection_t,
+    path: *const std::ffi::c_char,
+    entry_point: *const std::ffi::c_char,
+    out_err_msg: *mut *const std::ffi::c_char,
+) -> std::ffi::c_int {
+    if path.is_null() {
+        set_err_msg("Null path".to_string(), out_err_msg);
+        return 1;
+    }
+    let path = unsafe { std::ffi::CStr::from_ptr(path) };
+    let path = match path.to_str() {
+        Ok(path) => path,
+        Err(e) => {
+            set_err_msg(format!("Wrong path: {}", e), out_err_msg);
+            return 2;
+        }
+    };
+    let mut entry_point_option = None;
+    if !entry_point.is_null() {
+        let entry_point = unsafe { std::ffi::CStr::from_ptr(entry_point) };
+        entry_point_option = match entry_point.to_str() {
+            Ok(entry_point) => Some(entry_point),
+            Err(e) => {
+                set_err_msg(format!("Wrong entry point: {}", e), out_err_msg);
+                return 4;
+            }
+        };
+    }
+    if conn.is_null() {
+        set_err_msg("Null connection".to_string(), out_err_msg);
+        return 5;
+    }
+    let conn = conn.get_ref();
+    match RT.block_on(async move {
+        let _guard = LoadExtensionGuard::new(conn)?;
```

Comment:
Do we need to use guard here or we better to expose enable_load_extension(onoff: bool) method instead?

BTW, `LoadExtensionGuard` ignore potential error which can happen on `drop` of the guard during disable of `enable_extension` API

Reply:
I guess this is more of a comment to the PR that introduced it (https://github.com/tursodatabase/libsql/pull/1459) as the problem you're describing is in Rust API. I'm using the Rust API in a recommended way here. You have a fair point but I don't want to fix the Rust API in this PR. Please open an issue and let's fix the Rust API in separate PR.